tests/pipeline/test_quarters_estimates.py

```diff
@@ -0,0 +1,747 @@
+q2_knowledge_dates = [pd.Timestamp('2015-01-16'), pd.Timestamp('2015-01-20'),
+                      pd.Timestamp('2015-01-24'), pd.Timestamp('2015-01-28')]
+# We want to model the possibility of an estimate predicting a release date
+# that doesn't match the actual release. This could be done by dynamically
+# generating more combinations with different release dates, but that
+# significantly increases the amount of time it takes to run the tests. These
+# hard-coded cases are sufficient to know that we can update our beliefs when
+# we get new information.
+q1_release_dates = [pd.Timestamp('2015-01-15'),
+                    pd.Timestamp('2015-01-16')]  # One day late
+q2_release_dates = [pd.Timestamp('2015-01-30'),  # One day early
+                    pd.Timestamp('2015-01-31')]
+estimates = pd.DataFrame({
+    EVENT_DATE_FIELD_NAME: q1_release_dates + q2_release_dates,
+    'estimate': [.1, .2, .3, .4],
+    FISCAL_QUARTER_FIELD_NAME: [1.0, 1.0, 2.0, 2.0],
+    FISCAL_YEAR_FIELD_NAME: [2015.0, 2015.0, 2015.0, 2015.0]
+})
+
+
+def gen_estimates():
+    sid_estimates = []
+    sid_releases = []
+    for sid, (q1e1, q1e2, q2e1, q2e2) in enumerate(
+            itertools.permutations(q1_knowledge_dates + q2_knowledge_dates,
+                                   4)
+    ):
+        # We're assuming that estimates must come before the relevant release.
+        if (q1e1 < q1e2 and
+                q2e1 < q2e2 and
+                q1e1 < q1_release_dates[0] and
+                q1e2 < q1_release_dates[1]):
+            sid_estimate = estimates.copy(True)
+            sid_estimate[TS_FIELD_NAME] = [q1e1, q1e2, q2e1, q2e2]
+            sid_estimate[SID_FIELD_NAME] = sid
+            sid_estimates += [sid_estimate]
+            sid_release = releases.copy(True)
+            sid_release[SID_FIELD_NAME] = sid_estimate[SID_FIELD_NAME]
+            sid_releases += [sid_release]
+    return pd.concat(sid_estimates + sid_releases).reset_index(drop=True)
+
+
+class EstimateTestCase(object):
+    START_DATE = pd.Timestamp('2014-12-28')
+    END_DATE = pd.Timestamp('2015-02-03')
+
+    @classmethod
+    def make_loader(cls, events, columns):
+        pass
+
+    @classmethod
+    def init_class_fixtures(cls):
+        cls.sids = cls.events['sid'].unique()
+        cls.columns = {
+            Estimates.estimate: 'estimate',
+            Estimates.event_date: EVENT_DATE_FIELD_NAME,
+            Estimates.fiscal_quarter: FISCAL_QUARTER_FIELD_NAME,
+            Estimates.fiscal_year: FISCAL_YEAR_FIELD_NAME,
+        }
+        cls.loader = cls.make_loader(cls.events, cls.columns)
+
+        cls.ASSET_FINDER_EQUITY_SIDS = list(
+            cls.events[SID_FIELD_NAME].unique()
+        )
+        cls.ASSET_FINDER_EQUITY_SYMBOLS = [
+            's' + str(n) for n in cls.ASSET_FINDER_EQUITY_SIDS
+        ]
+        super(EstimateTestCase, cls).init_class_fixtures()
+
+    def _test_wrong_num_quarters_passed(self):
```

Comment:
we can drop the leading underscore and then tests that mix this in will inherit the test automatically.